src/tools/manifest2proto.ts

```diff
@@ -227,6 +227,7 @@ async function recipeToProtoPayload(recipe: Recipe) {
 
   const handleToProtoPayload = new Map<Handle, {name: string}>();
   for (const h of recipe.handles) {
+    h.type.maybeEnsureResolved();
```

Comment:
Please add a comment:
```
// After type inference which runs as a part of the recipe.normalize() above
// all handle types are constrained type variables. We force these type variables
// to their resolution by called maybeEnsureResolved(), so that handle types
// are encoded with concrete types, instead of variables.
```